packages/block-library/src/site-title/edit/index.js

```diff
@@ -34,7 +35,7 @@ export default function SiteTitleEdit( {
 		const siteData = getEntityRecord( 'root', '__unstableBase' );
 		return {
 			canUserEdit: canUser( 'update', 'settings' ),
-			readOnlyTitle: siteData?.name,
+			readOnlyTitle: siteData?.name && decodeEntities( siteData.name ),
```

Comment:
```suggestion
			readOnlyTitle: decodeEntities( siteData?.name ),
```

I think we can just pass data here no need for an extra check. The `decodeEntities` will return the passed value if it's not a string.